### PR TITLE
Merge attempt2

### PR DESCRIPTION
**Pull request by RafalJDev**

### CODE REVIEW

**Comment by gacikjakub** [comment]
Build SUCCESS.
Deployed on: http://mergeattempt2.battleships.kuba005.nazwa.pl/

File: backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class PlayerHolder {
+
+    private Set<Player> players;
+
+    public PlayerHolder() {
+        this.players = new HashSet<>();
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java]: new in the constructor?

File: backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public boolean addPlayer(Player playerDAO) {
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java]: Player playerDAO -> Player player

File: backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        if (playerDAO == null) {
+            throw new IllegalArgumentException("Player cannot be null");
+        }
+        System.out.println("Player " + playerDAO.getName() + " has been added to PlayerHolder");
+        return players.add(playerDAO);
+    }
+
+    public boolean isRegister(Player playerDAO) {
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java]: Player playerDAO -> Player player

File: backend/src/main/java/pl/krkteam/authentication/RegistrationController.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    public String signPlayer(@RequestBody String post) {
+        System.out.println(post);
+
+        Gson gson = new Gson();
+
+        PlayerDTO playerDTOFromFrontend = gson.fromJson(post, PlayerDTO.class);
+
+        System.out.println(playerDTOFromFrontend);
+
+        Player backendPlayerDAO = PlayerConversionUtil.convertDTOtoDAO(playerDTOFromFrontend);
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/authentication/RegistrationController.java]: Player or PlayerDAO?

File: backend/src/main/java/pl/krkteam/authentication/RegistrationController.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        boolean result = playerHolder.addPlayer(backendPlayerDAO);
+
+        String send = "{\"result\":" + "\"" + result + "\"" + " }";
+        return send;
+    }
+
+    @GetMapping(value = "/registered")
+    public String getPlayers() {
+        List<Player> daoPlayers = playerHolder.getPlayers();
+        PlayerDTO[] dtoPlayerDTOS = PlayerConversionUtil.convertDAOPlayersToDTO(daoPlayers);
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/authentication/RegistrationController.java]: playerDTOs

File: backend/src/main/java/pl/krkteam/authentication/RegistrationController.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+        boolean result = playerHolder.addPlayer(backendPlayerDAO);
+
+        String send = "{\"result\":" + "\"" + result + "\"" + " }";
+        return send;
+    }
+
+    @GetMapping(value = "/registered")
+    public String getPlayers() {
+        List<Player> daoPlayers = playerHolder.getPlayers();
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/authentication/RegistrationController.java]: players

File: backend/src/main/java/pl/krkteam/player/Player.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        return name;
+    }
+
+    private final String name;
+
+    public Player(String name) {
+        this.name = name;
+    }
+
+    @Override
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/player/Player.java:16]: Do override hashCode too

File: backend/src/main/java/pl/krkteam/player/PlayerConversionUtil.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package pl.krkteam.player;
+
+import java.util.Collection;
+
+public class PlayerConversionUtil {
+
+    public static PlayerDTO convertDAOtoDTO(Player playerDAO) {
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/player/PlayerConversionUtil.java]: convertPlayerToPlayerDTO
Player playerDAO -> Player player

File: backend/src/main/java/pl/krkteam/player/PlayerConversionUtil.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        PlayerDTO result = new PlayerDTO();
+        result.name = playerDAO.getName();
+        return result;
+    }
+
+    public static Player convertDTOtoDAO(PlayerDTO playerDTO) {
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/player/PlayerConversionUtil.java]: convertPlayerDTOToPlayer

File: backend/src/main/java/pl/krkteam/player/PlayerConversionUtil.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+        return new Player(playerDTO.name);
+    }
+
+    public static  PlayerDTO[] convertDAOPlayersToDTO(Collection<Player> playerCollection) {
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/player/PlayerConversionUtil.java]: remove double space

Comment (xKoem) [reply backend/src/main/java/pl/krkteam/player/PlayerConversionUtil.java]: convertPlayersToPlayerDTOs

File: backend/src/main/java/pl/krkteam/player/PlayerConversionUtil.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+        return playerCollection.stream()
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/player/PlayerConversionUtil.java]: :+1:

File: backend/src/main/java/pl/krkteam/player/PlayerDTO.java [hunk 12]
```diff
@@ -0,0 +1,21 @@
+package pl.krkteam.player;
+
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/player/PlayerDTO.java:2]: remove whitespace

File: frontend/src/app/models/board/row/cell/content/content.ts [hunk 13]
```diff
@@ -3,3 +3,25 @@ export interface Content {
   
   isShip(): boolean;
 }
+
+// export class ContentMiss implements Content{
```
Comment (xKoem) [inline frontend/src/app/models/board/row/cell/content/content.ts]: Comments in code :(

File: frontend/src/app/models/dto/ships/ships-dao.service.ts [hunk 14]
```diff
@@ -0,0 +1,8 @@
+import {Injectable} from "@angular/core";
+
+@Injectable()
+export class ShipsDaoService {
```
Comment (xKoem) [inline frontend/src/app/models/dto/ships/ships-dao.service.ts:4]: what does it do?

File: frontend/src/app/models/dto/shot/result/shot-result-factory.ts [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+import {ResultMiss} from "./result-miss";
+
+export class ShotResultFactory {
+  
+  static createShotFromJson(shotResultJson: HttpEvent<string>): ShotResult {
+    
+    const shotResultMessage = shotResultJson["shotResult"];
+    let shotResult: ShotResult;
+    
+    switch (shotResultMessage) {
```
Comment (xKoem) [inline frontend/src/app/models/dto/shot/result/shot-result-factory.ts]: Awesome switch <3 but no.

File: frontend/src/app/services/click-handler/oppent-board-handler.service.ts [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+  
+  handleShotClick(boardDiv: Element, rowIndex, columnIndex) {
+    console.log("click!");
+  
+    if (this.isPlayerTurn) {
+    
+      const resultHit = new ResultHit();
+      resultHit.handleResult(boardDiv, rowIndex, columnIndex);
+    
+      // this.shotSender.postShot("coordinate")
```
Comment (xKoem) [inline frontend/src/app/services/click-handler/oppent-board-handler.service.ts]: :(

File: frontend/src/app/services/click-handler/oppent-board-handler.service.ts [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+@Injectable()
+export class OpponentBoardHandler {
+  
+  isPlayerTurn: boolean = true;
+  
+  constructor(public shipSender: ShipSender, public shotSender: ShotSender) {
+  }
+  
+  handleShotClick(boardDiv: Element, rowIndex, columnIndex) {
+    console.log("click!");
```
Comment (xKoem) [inline frontend/src/app/services/click-handler/oppent-board-handler.service.ts]: :heart:

File: frontend/src/app/services/dom-manipulator/dom-manipulator.ts [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+    const cellDiv = this.getCellDiv(boardDiv, rowIndex, columnIndex);
+    const innerColorDiv = this.getInnerColorDiv(boardDiv, rowIndex, columnIndex);
+    
+    cellDiv.removeChild(innerColorDiv);
+    
+    let newInnerColorDiv = document.createElement('div');
+    newInnerColorDiv.classList.add("inner-color-hit");
+    newInnerColorDiv.classList.add("inner-color-locator");
+    
+    // newInnerColorDiv.setAttribute("width", "35px");
```
Comment (xKoem) [inline frontend/src/app/services/dom-manipulator/dom-manipulator.ts]: :(

File: frontend/src/app/services/dom-manipulator/dom-manipulator.ts [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+  public static setShip(boardDiv: Element, rowIndex, columnIndex) {
+    const shipDiv = this.getShipDiv(boardDiv, rowIndex, columnIndex);
+    
+    shipDiv.classList.add("ship1");
+  }
+  
+  public static colorCell(boardDiv: Element, rowIndex, columnIndex, color: string) {
+    const innerColor = this.getInnerColorDiv(boardDiv, rowIndex, columnIndex);
+    
+    console.log("I will color this cell");
```
Comment (xKoem) [inline frontend/src/app/services/dom-manipulator/dom-manipulator.ts]: more precise logs

File: frontend/src/app/services/drag-ship/drag-ship.service.ts [hunk 20]
```diff
@@ -27,26 +27,45 @@ export class DragShipService {
...
-      let square = this.getSquare(row, rowIndex, columnIndex);
+      let row = this.getRow(boardDiv, rowIndex);
+      let cell = this.getCell(row, rowIndex, columnIndex);
     
-      for (var k = 0; k < 5; k++) {
-        if (this.isThereNoShip(square)) {
-          innerDiv.classList.add("ship" + this.ship.mastCount);
+      let innerColor = this.getInnerColor(cell);
+  
+      // for (var k = 0; k < 5; k++) {
```
Comment (xKoem) [inline frontend/src/app/services/drag-ship/drag-ship.service.ts]: why this for is commented?

File: frontend/src/app/services/drag-ship/drag-ship.service.ts [hunk 21]
```diff
@@ -27,26 +27,45 @@ export class DragShipService {
...
+        if (this.isThereNoShip(cell)) {
+  
+          if (innerColor != null) {
+            // innerColor.classList.remove("inner-color");
+          innerColor.classList.add("inner-color-black");
+  
+            innerColor.setAttribute("width", "35px");
```
Comment (xKoem) [inline frontend/src/app/services/drag-ship/drag-ship.service.ts]: remove whitespace

File: frontend/src/app/services/drag-ship/drag-ship.service.ts [hunk 22]
```diff
@@ -27,26 +27,45 @@ export class DragShipService {
...
+    return innerSquare;
+  }
+  
+  private getCell(row: Element, rowIndex: string, columnIndex: string) {
     const squareWithNumber = "square" + rowIndex + columnIndex;
     let squareList: NodeListOf<Element> = row.getElementsByClassName(squareWithNumber);
     let square = squareList.item(0);
   
-    //square.setAttribute("color", "#00b3ee");
+    // square.setAttribute("color", "#00b3ee");
```
Comment (xKoem) [inline frontend/src/app/services/drag-ship/drag-ship.service.ts]: :(

File: frontend/src/app/views/boards/board/board.component.html [hunk 23]
```diff
@@ -1,28 +1,28 @@
-<script src="../../services/drag-ship/drag-ship.service.ts"></script>
-<div #ships
+<!--
```
Comment (xKoem) [inline frontend/src/app/views/boards/board/board.component.html]: Why this is commented?

File: frontend/src/app/views/boards/board/board.component.ts [hunk 24]
```diff
@@ -13,43 +12,35 @@ import {ShipGenerator} from "../../../services/ship-generator/ship-generator.ser
...
+  counter: number = 1;
   
-  putShipsFromShipGenerator_random() {
-    const shipArray = this.shipGenerator.generateShipsRandomly(this.board);
+  constructor(public dragShipService: DragShipService,
+              public shipGenerator: ShipGenerator) {
     
-    this.testShipSending(shipArray);
+    console.log("log:" + this.isOpponentBoard);
+    // if (!this.isOpponentBoard) {
```
Comment (xKoem) [inline frontend/src/app/views/boards/board/board.component.ts]: comments again

File: frontend/src/app/views/boards/opponent-board/opponent-board.component.css [hunk 25]
```diff
@@ -0,0 +1,30 @@
+.field {
+  display: inline-block;
+  font-size: 0;
+  margin: 10px 0;
+}
+
+.fieldBorder {
+  /*border-right: 20px solid #2B2D20;*/
```
Comment (xKoem) [inline frontend/src/app/views/boards/opponent-board/opponent-board.component.css]: :(

File: frontend/src/app/views/boards/opponent-board/opponent-board.component.html [hunk 26]
```diff
@@ -1,4 +1,24 @@
...
+     class="field col-12"
 >
+  <div *ngFor="let row of opponentBoard.board;let y = index"
+       class="row row{{y}}"
+  >
+    <div *ngFor="let cell of row.row; let x = index"
+         class="square square{{y}}{{x}} "
+         (click)="opponentBoardHandler.handleShotClick(boardDiv, y, x)"
+    >
+      <!--(click)="handleClick(boardDiv, y ,x)"-->
```
Comment (xKoem) [inline frontend/src/app/views/boards/opponent-board/opponent-board.component.html]: :(

File: frontend/src/app/views/boards/opponent-board/opponent-board.component.ts [hunk 27]
```diff
@@ -1,16 +1,98 @@
...
+  counter: number = 1;
+  
+  constructor(public opponentBoardHandler: OpponentBoardHandler) {
+    this.fillArrayWithEmpty();
+    
+    this.opponentBoard = new BoardOfCells();
+    
+    this.generateOpponentBoardWithWater();
+    
+    // if (!this.isOpponentBoard) {
```
Comment (xKoem) [inline frontend/src/app/views/boards/opponent-board/opponent-board.component.ts]: :(((((

File: frontend/src/app/views/boards/opponent-board/opponent-board.component.ts [hunk 28]
```diff
@@ -1,16 +1,98 @@
...
+      console.log("shotsArray" + this.shotsArray[rowIndex][columnIndex]);
+      console.log("get in");
+      return "black";
+    }
+    
+    // console.log("i won't be here");
+    console.log("color:" + this.squareColor);
+    return this.squareColor;
+    
+    // return this.clickedCoordinate.rowIndex == rowIndex && this.clickedCoordinate.columnIndex == columnIndex ?
```
Comment (xKoem) [inline frontend/src/app/views/boards/opponent-board/opponent-board.component.ts]: :(

File: frontend/src/app/views/boards/opponent-board/opponent-board.component.ts [hunk 29]
```diff
@@ -1,16 +1,98 @@
...
+    console.log("OpponentBoard:");
+    
+    console.log("indexes: " + rowIndex + " : " + columnIndex);
+    if (this.shotsArray[rowIndex][columnIndex] === 'X') {
+      console.log("shotsArray" + this.shotsArray[rowIndex][columnIndex]);
+      console.log("get in");
+      return "black";
+    }
+    
+    // console.log("i won't be here");
```
Comment (xKoem) [inline frontend/src/app/views/boards/opponent-board/opponent-board.component.ts]: :(

File: frontend/src/app/views/boards/player-board/player-board.component.html [hunk 30]
```diff
@@ -1,4 +1,31 @@
...
+  <div *ngFor="let row of playerBoard.board;let y = index"
+       class="row row{{y}}"
+  >
+    <div *ngFor="let cell of row.row; let x = index"
+         class="square square{{y}}{{x}} "
+         (dragover)="dragShipService.dragOver($event, boardDiv, y, x)"
+    >
 
-</app-board-component>
+      <!--[ngStyle]="{'background-color':  isOpponentBoard ? whatColor(y, x)  : squareColor}"-->
```
Comment (xKoem) [inline frontend/src/app/views/boards/player-board/player-board.component.html]: :(

File: frontend/src/app/views/boards/player-board/player-board.component.ts [hunk 31]
```diff
@@ -7,17 +12,48 @@ import {Component, EventEmitter, OnInit, Output} from '@angular/core';
            })
 export class PlayerBoardComponent implements OnInit {
   
-  @Output()
-  isFleetCorrectFromPlayer: EventEmitter<Boolean> = new EventEmitter<Boolean>();
+  //@Output()
```
Comment (xKoem) [inline frontend/src/app/views/boards/player-board/player-board.component.ts]: :(

File: frontend/src/app/views/boards/player-board/player-board.component.ts [hunk 32]
```diff
@@ -7,17 +12,48 @@ import {Component, EventEmitter, OnInit, Output} from '@angular/core';
...
+    
+    const shipArrayJson = JSON.stringify(shipArray);
+  
+    console.log(shipArrayJson);
+  
+    this.shipSender.postShip(shipArrayJson)
+        .then(result => {
+          this.result2 = result['result'];
+          if (this.result2 == 'true') {
+            // this.isFleetCorrectFromPlayer.emit(true);
```
Comment (xKoem) [inline frontend/src/app/views/boards/player-board/player-board.component.ts]: :(

File: frontend/src/app/views/configuration-screen/configuration-screen.component.html [hunk 33]
```diff
@@ -7,19 +7,21 @@
     <input type="text" matInput [(ngModel)]="player.name" placeholder="Player name:">
   </mat-input-container>
 
-  <mat-input-container class="col-12">
-    <input type="text" matInput [(ngModel)]="player.roomNumber" placeholder="Player room number:">
-  </mat-input-container>
+  <!--<mat-input-container class="col-12">-->
```
Comment (xKoem) [inline frontend/src/app/views/configuration-screen/configuration-screen.component.html]: :(

File: frontend/src/styles.css [hunk 34]
```diff
@@ -29,3 +29,16 @@
   background-size: 24px;
   background-image: url("assets/ship3.png");
 }
+
+.wowarship {
+  background: transparent url("assets/WoWarships.jpg");
+  /*z-index: -1;*/
```
Comment (xKoem) [inline frontend/src/styles.css]: :(

File: pom.xml [hunk 35]
```diff
@@ -13,7 +13,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <main.basedir>${project.basedir}</main.basedir>
+        <!--<main.basedir>${project.basedir}</main.basedir>-->
```
Comment (xKoem) [inline pom.xml]: :(

**Review by xKoem — CHANGES_REQUESTED** [review]

**Review by xKoem — APPROVED** [review]

File: backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java [hunk 36]
```diff
@@ -0,0 +1,33 @@
...
+
+public class PlayerHolder {
+
+    private Set<Player> players = new HashSet<>();;
+
+    public boolean addPlayer(Player player) {
+        if (player == null) {
+            throw new IllegalArgumentException("Player cannot be null");
+        }
+        System.out.println("Player " + player.getName() + " has been added to PlayerHolder");
```
Comment (agaped) [inline backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java:15]: please consider using external logger

File: backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java [hunk 37]
```diff
@@ -0,0 +1,33 @@
...
+
+    public boolean isRegister(Player player) {
+        if (player == null) {
+            throw new IllegalArgumentException("Player cannot be null");
+        }
+        return players.contains(player);
+    }
+
+    public List<Player> getPlayers() {
+        return new ArrayList<>(players);
```
Comment (agaped) [inline backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java]: consider returning collection view e.g. immutable view

File: backend/src/main/java/pl/krkteam/authentication/RegistrationController.java [hunk 38]
```diff
@@ -0,0 +1,46 @@
...
+
+    @PostMapping(value = "/login")
+    public String signPlayer(@RequestBody String post) {
+        System.out.println(post);
+
+        Gson gson = new Gson();
+
+        PlayerDTO playerDTO = gson.fromJson(post, PlayerDTO.class);
+
+        System.out.println(playerDTO);
```
Comment (agaped) [inline backend/src/main/java/pl/krkteam/authentication/RegistrationController.java:24]: should not be in the production code

File: frontend/src/app/views/boards/opponent-board/opponent-board.component.css [hunk 39]
```diff
@@ -0,0 +1,28 @@
+.field {
+  display: inline-block;
+  font-size: 0;
+  margin: 10px 0;
```
Comment (agaped) [inline frontend/src/app/views/boards/opponent-board/opponent-board.component.css:4]: consider adding flexible margin e.g.  10%, one css technics is a flexbox

File: backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java [hunk 40]
```diff
@@ -0,0 +1,33 @@
...
+import pl.krkteam.player.Player;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class PlayerHolder {
+
+    private Set<Player> players = new HashSet<>();;
```
Comment (agaped) [inline backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java]: redundant semicolon

**Review by agaped — CHANGES_REQUESTED** [review]
please correct

File: backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java (hunk 36 above)
Comment (gacikjakub) [reply backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java:15]: Refactor with logger will be done at the end of this iteration.

File: backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java (hunk 37 above)
Comment (gacikjakub) [reply backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java]: Done

File: backend/src/main/java/pl/krkteam/authentication/RegistrationController.java (hunk 38 above)
Comment (gacikjakub) [reply backend/src/main/java/pl/krkteam/authentication/RegistrationController.java:24]: Logger will be use at the end of this iteration

File: backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java (hunk 40 above)
Comment (gacikjakub) [reply backend/src/main/java/pl/krkteam/authentication/PlayerHolder.java]: done

File: frontend/src/app/views/boards/opponent-board/opponent-board.component.css (hunk 39 above)
Comment (gacikjakub) [reply frontend/src/app/views/boards/opponent-board/opponent-board.component.css:4]: done

**Review by agaped — APPROVED** [review]
:+1:

**Review by krzysieksr — APPROVED** [review]

**Pull request merged**